src/hyperv/vmbus/vmbus.c

```diff
@@ -429,12 +429,12 @@ static boolean vmbus_connect(vmbus_dev dev, uint32_t version)
         kernel_delay(milliseconds(1));
     }
 
-    int done = ((const struct vmbus_chanmsg_connect_resp *)
-        msg->msg_data)->chm_done;
+    const struct vmbus_chanmsg_connect_resp *connect_resp =
+        (const struct vmbus_chanmsg_connect_resp *)msg;
```

Comment:
It's `msg->msg_data`, not `msg`, that must be cast to `struct vmbus_chanmsg_connect_resp *`.

Reply:
Ugh, very careless of me. :( Thanks for catching these.